### PR TITLE
document jax.Array methods and attributes

### PR DESCRIPTION
**Pull request by jakevdp**
Preview of the newly-documented methods and attributes: https://jax--23080.org.readthedocs.build/en/23080/_autosummary/jax.Array.html#jax.Array

### CODE REVIEW

File: docs/jax.rst [hunk 1]
```diff
@@ -133,6 +133,74 @@ jax.Array (:code:`jax.Array`)
     make_array_from_single_device_arrays
     make_array_from_process_local_data
 
+Array properties and methods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. autosummary::
+  :toctree: _autosummary
+
+    Array.addressable_data
```
Comment (yashk2810) [inline docs/jax.rst:142]: Can we skip addressable_data please? I want to remove it

Comment (jakevdp) [reply docs/jax.rst:142]: Will do

File: jax/_src/array.py [hunk 2]
```diff
@@ -471,6 +481,7 @@ def __reduce__(self):
 
   @use_cpp_method()
   def unsafe_buffer_pointer(self):
```
Comment (yashk2810) [inline jax/_src/array.py:483]: Can we skip this too? We also want to remove this or not publicize it

File: jax/_src/array.py [hunk 3]
```diff
@@ -370,6 +379,7 @@ def __iter__(self):
 
   @property
   def is_fully_replicated(self) -> bool:
+    """Return True if the array is fully replicated across its shards."""
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/array.py]: Remove this because the relevant docs are already in `basearray.py`

**Review by yashk2810 — APPROVED** [review]
Looks like array.py vanished. Was that intentional?

**Comment by jakevdp** [comment]
> Looks like array.py vanished. Was that intentional?

Yes – the docstrings there don't matter because docs are pulled from methods in `basearray.py`.

**Review by froystig — APPROVED** [review]

**Pull request merged**